hydromt_wflow/wflow.py

```diff
@@ -1103,18 +1119,88 @@ def setup_lakes(self, lakes_fn="hydro_lakes", min_area=10.0):
         * **LakeAvgLevel** map: lake average water level [m]
         * **LakeAvgOut** map: lake average discharge [m3/s]
         * **Lake_b** map: lake rating curve coefficient [-]
+        * **LakeOutflowFunc** map: option to compute rating curve [-]
+        * **LakeStorFunc** map: option to compute storage curve [-]
+        * **LakeMaxStorage** map: optional, maximum storage of lake [m3]
         * **lakes** geom: polygon with lakes and wflow lake parameters
 
         Parameters
         ----------
-        lakes_fn : {'hydro_lakes'}
+        lakes_fn :
             Name of data source for lake parameters, see data/data_sources.yml.
 
-            * Required variables: ['waterbody_id', 'Area_avg', 'Vol_avg', 'Depth_avg', 'Dis_avg']
+            * Required variables for direct use: ['waterbody_id', 'Area_avg', 'Depth_avg', 'Dis_avg', 'Lake_b', 'Lake_e', 'LakeOutflowFunc', 'LakeStorFunc', 'LakeThreshold', 'LinkedLakeLocs']
+
+            * Required variables for parameter estimation: ['waterbody_id', 'Area_avg', 'Vol_avg', 'Depth_avg', 'Dis_avg']
+        rating_curve_fns: str, Path, List[str], List[Path], optional
+            Data catalog entry/entries or path(s) containing rating curve values for lakes. If None then will be derived from
+            properties of lakes_fn. Assumes one file per lake (with all variables) and that the lake ID is either in the filename
+            or data catalog entry name (eg using placeholder). The ID should be placed at the end separated by an underscore (eg
+            'rating_curve_12.csv' or 'rating_curve_12')
+
+            * Required variables: ['elevtn', 'volume'] for storage curve and ['elevtn', 'discharge'] for discharge rating curve
         min_area : float, optional
-            Minimum lake area threshold [km2], by default 1.0 km2.
+            Minimum lake area threshold [km2], by default 10.0 km2.
+        add_maxstorage : bool, optional
+            If True, maximum storage of the lake is added to the output (controlled lake) based on 'Vol_max' [m3] column of lakes_fn, by default False (natural lake).
         """
 
+        # Derive lake are and outlet maps
+        gdf_org, ds_lakes = self._setup_waterbodies(lakes_fn, "lake", min_area)
+        if ds_lakes is None:
+            return
+        rmdict = {k: v for k, v in self._MAPS.items() if k in ds_lakes.data_vars}
+        ds_lakes = ds_lakes.rename(rmdict)
+
+        # If rating_curve_fn prepare rating curve dict
+        rating_dict = dict()
+        if rating_curve_fns is not None:
+            rating_curve_fns = np.atleast_1d(rating_curve_fns)
+            # Find ids in rating_curve_fns
+            fns_ids = [int(fn.split("_")[-1].split(".")[0]) for fn in rating_curve_fns]
```

Comment:
Would a try-except statement with a meaningfull error message help here in case the ids cannot be parsed as integers from the filenames